generator/testdata/google/example/library/v1/library_service.gapic.cc.baseline

```diff
@@ -9,84 +11,192 @@
 gax::StatusOr<::google::example::library::v1::Book>
 LibraryService::CreateBook(
 ::google::example::library::v1::CreateBookRequest const& request) {
-  // TODO: actual useful work, e.g. retry, backoff, metadata, pagination, etc.
+  // TODO: update metadata and idempotency, etc.
   grpc::ClientContext context;
   ::google::example::library::v1::Book response;
   grpc::Status status = stub_->CreateBook(&context, request, &response);
+
   if (status.ok()) {
     return response;
-  } else {
-    return gax::GrpcStatusToGaxStatus(status);
   }
+
+  std::string error_message;
+  // TODO: use idempotency policy to determine whether a failure is permanent
+  if (!isPermanentFailure(status)) {
+    auto retry_policy = retry_policy_->clone();
+    auto backoff_policy = backoff_policy_->clone();
+    while (retry_policy->OnFailure(status)) {
+      auto backoff_duration = backoff_policy->OnCompletion();
+      std::this_thread::sleep_for(backoff_duration);
+      status = stub_->CreateBook(&context, request, &response);
```

Comment:
Every call to the stub requires a brand new grpc::ClientContext object - they can't be reused between RPCs.

Reply:
That was something I was unsure about; the documentation indicates that a new ClientContext is needed for every RPC, but engaging in retry seemed like the `same` RPC, from one point of view at least.

Reply:
gRPC will fail if you reuse the same ClientContext.  And there is no `clear()` or `reset()` function either.